internal/verifier/verifier.go

```diff
@@ -146,6 +146,38 @@ func (verifier *ProvenanceMetadataVerifier) Verify() (VerificationResult, error)
 	return provenanceVerifier.Verify()
 }
 
+// ValidatedProvenanceValues contain exactly one or no value in every field.
+type ValidatedProvenanceValues = ProvenanceIR
+
+// ValidatedReferenceValues contain at least 1 value for every field the provenanceFormat mandates to check,
+// and 0 values for every field that is to be ignored.
+type ValidatedReferenceValues = ProvenanceIR
+
+// If a new field is added to `ProvenanceIR`, this has to be updated.
+func (want *ProvenanceIR) ValidateReferenceValues(provenanceFormat string) (*ValidatedReferenceValues, error) {
```

Comment:
Where does `provenanceFormat` come from? 
All provenance formats are slsa provenances, with different buildTypes. I suggest we use the buildType as the format, and have a reference value for the supported buildTypes as well. WDYT @mariaschett?

Reply:
> Where does provenanceFormat come from?

Options: 
1. We infer the provenanceFormat from the given provenance.
2. The caller decides, which implicitly happens currently, when calling `ProvenanceMetadataVerifier` (we now that this is `SLSAv02`).

> buildType as format / reference value for the buildTypes

Let's discuss this offline.